resources/machine_name.rb

```diff
@@ -4,54 +4,73 @@
 property :computer_name, String, desired_state: true
 property :local_hostname, String, desired_state: true, coerce: proc { |name| conform_to_rfc1034(name) }
 property :netbios_name, String, desired_state: false, coerce: proc { |name| conform_to_rfc1034(name)[0, 15].upcase }
+property :dns_domain, String, desired_state: false, default: ''
 
 load_current_value do
-  hostname get_name('HostName')
+  hostname current_hostname
+  dns_domain current_dns_domain
+  netbios_name shell_out('defaults', 'read', '/Library/Preferences/SystemConfiguration/com.apple.smb.server.plist', 'NetBIOSName').stdout.chomp
   computer_name get_name('ComputerName')
   local_hostname get_name('LocalHostName')
 end
 
 action :set do
+  property_is_set?(:netbios_name) ? new_resource.netbios_name : new_resource.netbios_name = new_resource.hostname
+  property_is_set?(:computer_name) ? new_resource.computer_name : new_resource.computer_name = new_resource.hostname
+  property_is_set?(:local_hostname) ? new_resource.local_hostname : new_resource.local_hostname = new_resource.hostname
+
   converge_if_changed :hostname do
-    converge_by "set Hostname to #{new_resource.hostname}" do
-      execute [scutil, '--set', 'HostName', new_resource.hostname] do
+    converge_by 'set Hostname' do
+      fqdn = [new_resource.hostname, new_resource.dns_domain].join '.'
```

Comment:
This appends the `HostName` with a `.` at the end if the `dns_domain` property is not set.

Reply:
Good catch, thanks.